apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.CopyObject.cs

```diff
@@ -76,6 +76,10 @@ public sealed partial class StorageClientImpl : StorageClient
             options?.ModifyRequest(request);
             ApplyEncryptionKey(options?.EncryptionKey, options?.KmsKeyName, request);
             request.ModifyRequest += (options?.SourceEncryptionKey ?? EncryptionKey).ModifyRequestForRewriteSource;
+            if(options.IfGenerationMatch != null)
```

Comment:
Nit: space after `if`.

Reply:
Currently this will throw if `options` is null. I suggest you move the logic to the `ModifyRequest` method.